### ISSUE TITLE
Response times with ddcutil are slow - use DDCControl DBUS service if available

### ISSUE THREAD

**Issue by slaclau** [daitj/gnome-display-brightness-ddcutil#137]
ddcutil is slow as it opens a new DDC connection each time it's called, DDCControl provides a DBUS service which maintains the connection.

**Comment by daitj** [comment]
I can see that how having a service with open connection would benefit the core idea of this extension, to change brightness and fast. I haven't tested anything yet, but if DDCControl does provide faster get/set vcp actions, I cannot decide which option is better.

1. Rewrite everything with ddccontrol as the only requirement.
2. Write extra functions and settings to switch between ddccontrol or ddcutil
3. Make a new extension (same as 1) but leave this as it is for people wishing to stay with ddcutil

Have you been already working on that or is it just something you thought of? I can look this on my free time. 

Just something I noticied, 
ddcutil relies on ddc/ci standard and is very actively maintained by the main contributor
whereas ddccontrol relies on monitor database and isn't maintained as much as ddcutil.

**Comment by SebastJava** [comment]
Another option could be the `ddcutil-service`. Maybe, I don't know...

I was looking at https://extensions.gnome.org/extension/7308/displays-adjustments/ In there, it says:

> ddcutil-service allows more responsive communication with the displays than ddcutil.

I tried this Display Adjustment by w8jcik... It didn't work for some reason. Still, maybe it would be worth investigating?

**Comment by SebastJava** [comment]
I just looked at the "Advanced Settings", and changed the "Sleep Multiplier ms" to 4 (minimum value). And I changed the "Step Change %" to 1. It is now more responsive and accurate. Pretty good.
![image](https://github.com/user-attachments/assets/6d558635-f079-45c3-90dc-ff8913027b86)
But that's not an universal solution. What's good for my monitor could be bad on others.

I looked into https://www.ddcutil.com/performance_options/

> There are two ways to adjust wait times: explicitly using option --sleep-multiplier, and automatically using dynamic sleep adjustment.

The dynamic sleep adjustment looks like a better solution, at first sight. Easy and universal. All you would need to do is to remove that `--sleep-multiplier` option. The default is `--enable-dynamic-sleep`.

That's just my first impressions. I am not an expert...

**Comment by daitj** [comment]
Yeah `--enable-dynamic-sleep` along with `--skip-ddc-checks` are new options added after ddcutil 2+, I suppose soon it will be safe to use that, at least `--skip-ddc-checks`. 

But often in many optimization discussions rockowitz (ddcutil's developer) has suggested to find a right value for `sleep-multiplier` and use that.
